test/helpers/context-test-helpers.js

```diff
@@ -0,0 +1,67 @@
+var traverse = require('traverse');
+
+exports.ContextRecorder = ContextRecorder;
+exports.deepCloneToObject = deepCloneToObject;
+exports.aCtxForModel = aCtxForModel;
+
+function ContextRecorder(initialValue) {
+  if (!(this instanceof ContextRecorder)) {
+    return new ContextRecorder(initialValue);
+  }
+  this.records = initialValue;
+};
+
+ContextRecorder.prototype.recordAndNext = function(transformFm) {
+  var self = this;
+  return function(context, next) {
+    if (typeof transformFm === 'function') {
+      transformFm(context);
+    }
+
+    context = deepCloneToObject(context);
+    context.hookState.test = true;
+
+    if (typeof self.records === 'string') {
+      self.records = context;
+      return next();
+    }
+
+    if (!Array.isArray(self.records)) {
+      self.records = [self.records];
+    }
+
+    self.records.push(context);
+    next();
+  };
+};
+
+function deepCloneToObject(obj) {
+  return traverse(obj).map(function(x) {
+    if (x === undefined) {
+      // RDBMSs return null
+      return null;
+    }
+    if (x && x.toObject)
+      return x.toObject(true);
+    if (x && typeof x === 'function' && x.modelName)
+      return '[ModelCtor ' + x.modelName + ']';
+  });
+}
+
+function aCtxForModel(TestModel, ctx) {
+  ctx.Model = TestModel;
+
+  if (!ctx.hookState) {
+    ctx.hookState = {};
+  }
+
+  if (!('test' in ctx.hookState)) {
+    ctx.hookState.test = true;
+  }
+
+  if (!ctx.options) {
+    ctx.options = {};
+  }
+  return deepCloneToObject(ctx);
+}
+
```

Comment:
This is lame, I would expect `eslint` to catch this problem. I suppose we should turn on http://eslint.org/docs/rules/no-multiple-empty-lines.